### PR TITLE
Add lambda methods. 

### PR DESCRIPTION
**Pull request by damacus**
Now throws an error message. Useful if it can't find the lambda (e.g. you're pointing to the wrong region)

### CODE REVIEW

**Comment by damacus** [comment]
This "new" way of finding the lambdas will trigger a smaller response payload. 
`find_lambda` returns a list of functions. But still gets called on each search of a lambda. 
`lambda_config` still gets called on every run, but should be quicker for the system to parse, given it's returning a fraction of the information.

**Comment by damacus** [comment]
@k1LoW is it worth looking at changing the generator for this one to match the tests, or... Just go with what was there before?

**Comment by k1LoW** [comment]
I think "`list_functions` return values (via `find_lambda`)" and "`get_function_configuration` return values" is same.
If you use lambda resource type, find_lambda has always been called once. So I think that it is good as it is now implemented.

| function | call | values |
| -- | -- | -- |
| `list_functions` | Once in one type | http://docs.aws.amazon.com/sdkforruby/api/Aws/Lambda/Client.html#list_functions-instance_method |
| `get_function_configuration` | every by `its` called | http://docs.aws.amazon.com/sdkforruby/api/Aws/Lambda/Client.html#get_function_configuration-instance_method |

**Comment by damacus** [comment]
Either way, I couldn't get at the values of timeout and memory size with the current implementation. I'm not sure why really. 

I'm aware the code is pretty verbose, but it also should be, easy to read/maintain going forward, and importantly does actually work when I tested this locally yesterday 😊

**Comment by damacus** [comment]
When you say `Once in one type`, where is that information stored so I can get at it?

**Comment by k1LoW** [comment]
> where is that information stored so I can get at it?

https://github.com/k1LoW/awspec/pull/300/files#diff-8f98399a072eea5474dd08bdde7c7b30L4

File: lib/awspec/type/lambda.rb [hunk 1]
```diff
@@ -9,7 +9,55 @@ def id
     end
 
     def timeout
-      resource_via_client.timeout
```
Comment (damacus) [inline lib/awspec/type/lambda.rb]: This timeout methods wasn't available when running rspec yesterday (before I started hacking things around) 👍

**Pull request closed without merging**